src/ucp/proto/proto_am.inl

```diff
@@ -174,6 +174,8 @@ ucs_status_t ucp_do_am_zcopy_single(uct_pending_req_t *self, uint8_t am_id,
         req->send.state = saved_state; /* need to restore the offsets state */
         return status;
     } else {
+        ++req->send.uct_comp.count;
```

Comment:
Maybe better to unite dt_state and uct_comp to new req_state since they are updated synchronously and implement `req_state_set` (init and restore) and `req_state_advance` functions? Currently, they are ++ and -- separately, but it looks not good for me. I tried to keep them together for easier refactoring.